From 08483aefe94ea53802b9bd20f2d77d1da69ccaf0 Mon Sep 17 00:00:00 2001
From: Wei Yongjun <yongjun_wei@trendmicro.com.cn>
Date: Tue, 18 Jun 2013 11:40:55 +0800
Subject: [PATCH 0648/1302] usb: gadget: f_subset: fix missing unlock on error
 in geth_alloc()

Add the missing unlock before return from function geth_alloc()
in the error handling case.

Introduced by commit 02832e56f88a981474ee4c7c141f46fc1b4454f4.
(usb: gadget: f_subset: add configfs support)

Signed-off-by: Wei Yongjun <yongjun_wei@trendmicro.com.cn>
Acked-by: Andrzej Pietrasiewicz <andrzej.p@samsung.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>
Signed-off-by: MyungJoo Ham <myungjoo.ham@samsung.com>
---
 drivers/usb/gadget/f_subset.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/usb/gadget/f_subset.c b/drivers/usb/gadget/f_subset.c
index fbc7a24..5601e1d 100644
--- a/drivers/usb/gadget/f_subset.c
+++ b/drivers/usb/gadget/f_subset.c
@@ -548,6 +548,7 @@ static struct usb_function *geth_alloc(struct usb_function_instance *fi)
 					  sizeof(geth->ethaddr));
 	if (status < 12) {
 		kfree(geth);
+		mutex_unlock(&opts->lock);
 		return ERR_PTR(-EINVAL);
 	}
 	geth_string_defs[1].s = geth->ethaddr;
-- 
1.8.3.2

